### PR TITLE
Improve SoapClient

### PR DESCRIPTION
**Pull request by qw3ry**

### CODE REVIEW

File: src/main/java/sirius/kernel/xml/Outcall.java [hunk 1]
```diff
@@ -170,6 +198,23 @@ public void setAuthParams(String user, String password) throws IOException {
         }
     }
 
+    /**
+     * Makes the underlying connection trust self-signed certs.
+     *
+     * This will make the connection trust <strong>only</strong> self-signed certificates!
+     */
+    public void trustSelfSignedCertificates() {
```
Comment (andyHa) [inline src/main/java/sirius/kernel/xml/Outcall.java:206]: return Outcall?

File: src/main/java/sirius/kernel/xml/Outcall.java [hunk 2]
```diff
@@ -170,6 +198,23 @@ public void setAuthParams(String user, String password) throws IOException {
         }
     }
 
+    /**
+     * Makes the underlying connection trust self-signed certs.
+     *
+     * This will make the connection trust <strong>only</strong> self-signed certificates!
```
Comment (andyHa) [inline src/main/java/sirius/kernel/xml/Outcall.java:204]: there is no option to accept both, self-signed and proper SSL certs?

**Review by andyHa — COMMENTED** [review]
...

File: src/main/java/sirius/kernel/xml/Outcall.java (hunk 2 above)
Comment (qw3ry) [reply src/main/java/sirius/kernel/xml/Outcall.java:204]: Currently, no. I could trust all certificates instead. There is no other easy possibility.
However, you can always add a certificate manually to the JVM Truststore, then you don't need this option at all.

**Review by andyHa — APPROVED** [review]

File: src/main/java/sirius/kernel/xml/Outcall.java [hunk 3]
```diff
@@ -46,8 +55,26 @@
...
     private static final Pattern CHARSET_PATTERN = Pattern.compile("(?i)\\bcharset=\\s*\"?([^\\s;\"]*)");
+    private static final X509TrustManager TRUST_SELF_SIGNED_CERTS = new X509TrustManager() {
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+            throw new CertificateException("This trust manager cannot be used in a server");
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+            if (chain.length != 1) {
```
Comment (bwiedmann) [inline src/main/java/sirius/kernel/xml/Outcall.java:66]: guter Ansatz, bei self signed Certs hast ja keine CA die in der Kette drüber hängt 👍

**Review by bwiedmann — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by bwiedmann — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Review by jmuscireum — APPROVED** [review]

**Pull request merged**